nativelib/src/main/scala/scala/scalanative/runtime/ieee754tostring/ryu/RyuDouble.scala

```diff
@@ -0,0 +1,1108 @@
+// Copyright 2018 Ulf Adams
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Credits:
+//
+//    1) This work is a heavily modified derivation of the original work by
+//       Ulf Adams at URL: https://github.com/ulfjack/ryu.
+//       As such, it inherits the Apache license of the original work.
+//       Thank you Ulf Adams.
+//
+//    2) The original java sources were converted to a rough draft of
+//       scala code using the service at URL: javatoscala.com.
+//
+//       The raw conversion did not compile and contained bugs due
+//       to the handling of break and return statements, but it saved
+//       days, if not weeks, of effort.
+//
+//       Thank you javatoscala.com.
+//
+//    3) All additional work, including introduced bugs,  is an original
+//       contribution to Scala Native development.
+
+// Porting Note for POW5_SPLIT & POW5_INV_SPIT arrays.
+//
+// Enquiring minds want to know, rightly, how the values in these
+// two arrays were derived.  The process was largely manual and not easy.
+// The length of this note should be the first clue.
+//
+// The algorithm for the values comes from the Ryu GitHub RyuDouble.java
+// original. That file generates the values at each and every object
+// creation time. This code is performance sensitive.  Arrays of known
+// constants avoid that initialization cost.
+//
+// The .java code used two dimensional arrays. @densh recommended using one
+// dimensional arrays because 1D 'arrays of constant values are relatively.
+// compact (4 bytes per element)'.
+//
+// The values were generated by inserting printf() statements into
+// a private development copy of an early iteration of RyuDouble.scala
+// which initialized the arrays at object creation time.  The output
+// was captured into a file. Properly chosen printfs yielded the basic
+// comma separated 4 column format.
```

Comment:
Please provide a small self-contained program that prints the output of the POW_* constants here. Otherwise it's going to be hard to reproduce the code-generated parts based on textual description.

Reply:
Both RyuFloat and RyuDouble files need their own copy that prints corresponding arrays.

Reply:
Understood & agreed.  I will keep it as small as possible. To allow people reading the
file to get to the good stuff as soon as possible, I probably put a one line reference 
at the array declaration sites to the commented out program at the bottom of
each file.  I am less concerned with generator program file size if it at the
bottom of the file.   Acceptable?

For such a large comment, do you want // on every line or /* */ with * on each intervening line?

Does the `final` on the array of constants declarations add any value or just
reveal my lack of scala understanding.  I  think I had convinced myself that
the latter was true. The array is mutable with or without and I am not sure
why anybody would want to override in a sub-class.

Reply:
We generally use `/* */` for doc comments and `//` for everything else.

Reply:
`final` on array constants doesn't really do anything to best of my knowledge.